src/qibolab/instruments/qm.py

```diff
@@ -580,8 +580,11 @@ def fetch_results(result, ro_pulses):
             if f"{serial}_shots" in handles:
                 shots = handles.get(f"{serial}_shots").fetch_all().astype(int)
             else:
-                shots = None
-            results[pulse.qubit] = results[serial] = ExecutionResults.from_components(ires, qres, shots)
+                results[pulse.qubit] = results[serial] = (
+                    AveragedResults.from_components(ires, qres)
+                    if average
+                    else ExecutionResults.from_components(ires, qres, shots)
```

Comment:
Wait... there is something weird...

We're in the wrong branch of the `if`, where is going to take the `shots` from? The previous iteration?

Reply:
Actually, reading the history of comments, this is the same one of @stavros11 above https://github.com/qiboteam/qibolab/pull/346#discussion_r1149016470, and it has not actually been fixed by a14f998

Reply:
Thanks for catching this. It should be fixed by https://github.com/qiboteam/qibolab/pull/346/commits/2f426bbce2c67fadc9daa0beea740d9c7d5f54b3.